deepchem/models/torch_models/ferminet.py

```diff
@@ -244,24 +246,28 @@ def calculate_kinetic_energy(self,):
         """
         # using functorch to calcualte hessian and jacobian in one go
         # using index tensors to index out the hessian elemennts corresponding to the same variable (cross-variable derivatives are ignored)
-        i = torch.arange(self.batch_size).view(self.batch_size, 1, 1, 1, 1, 1,
-                                               1)
+        i = torch.arange(self.batch_size).view(self.batch_size, 1, 1, 1, 1)
         j = torch.arange(self.total_electron).view(1, self.total_electron, 1, 1,
-                                                   1, 1, 1)
-        k = torch.arange(3).view(1, 1, 3, 1, 1, 1, 1)
-
+                                                   1)
+        k = torch.arange(3).view(1, 1, 3, 1, 1)
         # doing all the calculation and detaching from graph to save memory, which allows larger batch size
-        jacobian_square_sum = torch.sum(torch.sum(torch.sum(torch.pow(
-            torch.func.jacrev(lambda x: torch.log(torch.abs(self.forward(x))))(
-                self.input), 2),
-                                                            axis=-1),
-                                                  axis=-1),
-                                        axis=-1).detach()
-        hessian_sum = torch.sum(torch.reshape(
-            torch.func.hessian(lambda x: torch.log(torch.abs(self.forward(x))))(
-                self.input)[i, i, j, k, i, j, k],
-            (self.batch_size, self.total_electron, 3)).detach(),
-                                axis=(1, 2))
+        # cloning self.input which will serve as the new input for the vectorized functions.
+        input = torch.clone(self.input).detach()
+        # lambda function for calculating the log of absolute value of the wave function.
+        # using jacrev for the jacobian and jacrev twice for to calculate the hessian. The functorch's hessian function if directly used does not give stable results.
+        jac = torch.func.jacrev(lambda x: torch.log(torch.abs(self.forward(x))))
+        hess = torch.func.jacrev(jac)
```

Comment:
Using recursive jacrev call(twice calling) to calculate the hessian. Functorch implementation of hessian is jacfwd(jacrev) but this does not cover all the operators presently for automatic differentiation.

Reply:
I also tried implementing vhp vectorized function, but right now vhp has bugs and cannot be used to called if jacobians calls were made earlier.

Implementing vhp in the future would decrease the load further by a very significant margin, as we need to calculate only the diagonal elements of the hessian (meaning double derivative wrt to the same variable).